src/anyio/_backends/_asyncio.py

```diff
@@ -416,6 +416,14 @@ def cancel_called(self) -> bool:
     def shield(self) -> bool:
         return self._shield
 
+    @shield.setter
+    def shield(self, value: bool) -> None:
+        if self._shield != value:
+            self._shield = value
+            # start cancellation effort if necessary when shield is removed
+            if not value and self._cancel_called and self._cancel_handle is None:
```

Comment:
The logic here is wrong because we want to restart the relevant cancellation delivery effort in a _parent_ cancel scope (may not be the immediate parent). Restarting it in the _current_ scope is pointless because shielding doesn't protect from direct cancellation anyway.